### PR TITLE
feat(j-s): Return indictment messages

### PR DESCRIPTION
**Pull request by unakb**
[Tilkynning til ákæranda um endursendingu](https://app.asana.com/0/1199153462262248/1206887069400347)

## What

Send a notification to the prosecutor when a judge returns a case

## Why

To alert the prosecutor

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Enhanced notification system to support notifications for returned indictments in judicial cases, improving communication with stakeholders.

- **Bug Fixes**
	- Optimized message handling logic for different case states and types to ensure precise notification delivery.

- **Refactor**
	- Introduced new methods for handling indictment-related scenarios, enhancing code efficiency and readability.

- **Documentation**
	- Updated internal documentation to include details on new notification types and processes.

### CODE REVIEW

File: apps/judicial-system/backend/src/app/modules/notification/notification.service.ts [hunk 1]
```diff
@@ -1538,6 +1538,39 @@ export class NotificationService {
...
+    )
+
+    return this.recordNotification(
+      theCase.id,
+      NotificationType.INDICTMENT_RETURNED,
+      [recipient],
+    )
+  }
+  //#endregion
   //#region Appeal notifications
```
Comment (unakb) [inline apps/judicial-system/backend/src/app/modules/notification/notification.service.ts:1574]:
```suggestion

  //#region Appeal notifications
```

File: apps/judicial-system/backend/src/app/modules/case/case.service.ts [hunk 2]
```diff
@@ -959,33 +974,37 @@ export class CaseService {
...
-        if (isIndictmentCase(updatedCase.type)) {
+        if (isIndictment) {
           await this.addMessagesForCompletedIndictmentCaseToQueue(
             updatedCase,
             user,
           )
         } else {
           await this.addMessagesForCompletedCaseToQueue(updatedCase, user)
         }
+      } else if (updatedCase.state === CaseState.SUBMITTED && isIndictment) {
```
Comment (unakb) [inline apps/judicial-system/backend/src/app/modules/case/case.service.ts:985]: I'd like to refactor this when we have time, all the if cases are getting a bit hard to read.

File: apps/judicial-system/backend/src/app/modules/case/case.service.ts [hunk 3]
```diff
@@ -1019,7 +1038,10 @@ export class CaseService {
     }
 
     if (updatedCase.courtCaseNumber) {
-      if (updatedCase.courtCaseNumber !== theCase.courtCaseNumber) {
+      if (
+        updatedCase.courtCaseNumber !== theCase.courtCaseNumber &&
+        updatedCase.courtCaseNumber !== ''
```
Comment (gudjong) [inline apps/judicial-system/backend/src/app/modules/case/case.service.ts:1043]: Please consider removing this line as this should be unnecessary. The empty string is falsy and we already checked that updatedCase.courtCaseNumber is falsy above.

File: apps/judicial-system/backend/src/app/modules/case/case.service.ts (hunk 2 above)
Comment (gudjong) [reply apps/judicial-system/backend/src/app/modules/case/case.service.ts:985]: Agree. We should also speed this up by waiting in parallel wherever posible.

**Review by gudjong — APPROVED** [review]
Minor comments

**Pull request merged**